include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2251 @@
+ *
+ * \param commandBuffer a command buffer
+ * \param source the texture region to copy from
+ * \param destination the texture region to copy to
+ * \param filterMode the filter mode that will be used when blitting
+ * \param cycle if SDL_TRUE, cycles the destination texture if the destination texture slice is bound, otherwise overwrites the data.
+ *
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuBlit(
+    SDL_GpuCommandBuffer *commandBuffer,
+    SDL_GpuTextureRegion *source,
+    SDL_GpuTextureRegion *destination,
+    SDL_GpuFilter filterMode,
+    SDL_bool cycle);
+
+/* Submission/Presentation */
+
+/**
+ * Obtains whether or not a swapchain composition is supported by the GPU backend.
+ *
+ * \param device a GPU context
+ * \param window an SDL_Window
+ * \param swapchainComposition the swapchain composition to check
+ *
+ * \returns SDL_TRUE if supported, SDL_FALSE if unsupported (or on error)
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_GpuSupportsSwapchainComposition(
+    SDL_GpuDevice *device,
+    SDL_Window *window,
+    SDL_GpuSwapchainComposition swapchainComposition);
+
+/**
+ * Obtains whether or not a presentation mode is supported by the GPU backend.
+ *
+ * \param device a GPU context
+ * \param window an SDL_Window
+ * \param presentMode the presentation mode to check
+ *
+ * \returns SDL_TRUE if supported, SDL_FALSE if unsupported (or on error)
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_GpuSupportsPresentMode(
+    SDL_GpuDevice *device,
+    SDL_Window *window,
+    SDL_GpuPresentMode presentMode);
+
+/**
+ * Claims a window, creating a swapchain structure for it.
+ * This must be called before SDL_GpuAcquireSwapchainTexture is called using the window.
+ *
+ * This function will fail if the requested present mode or swapchain composition
+ * are unsupported by the device. Check if the parameters are supported via
+ * SDL_GpuSupportsPresentMode / SDL_GpuSupportsSwapchainComposition prior to
+ * calling this function.
+ *
+ * SDL_GPU_PRESENTMODE_VSYNC and SDL_GPU_SWAPCHAINCOMPOSITION_SDR are
+ * always supported.
+ *
+ * \param device a GPU context
+ * \param window an SDL_Window
+ * \param swapchainComposition the desired composition of the swapchain
+ * \param presentMode the desired present mode for the swapchain
+ *
+ * \returns SDL_TRUE on success, otherwise SDL_FALSE.
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuAcquireSwapchainTexture
+ * \sa SDL_GpuUnclaimWindow
+ * \sa SDL_GpuSupportsPresentMode
+ * \sa SDL_GpuSupportsSwapchainComposition
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_GpuClaimWindow(
+    SDL_GpuDevice *device,
+    SDL_Window *window,
+    SDL_GpuSwapchainComposition swapchainComposition,
+    SDL_GpuPresentMode presentMode);
```

Comment:
How does this work in multi-GPU systems where the GPU you've selected isn't the one handling the display?

Reply:
ClaimWindow sets up the swapchain structure, which is mediated between the OS and the graphics driver. We don't have any more direct control over that and it hasn't been an issue so far.